### PR TITLE
Add standard interface for drivetrain

### LINKED ISSUE #1: Add standard interface for drivetrain
**Is your feature request related to a problem? Please describe.**
No.

**Describe the solution you'd like**
Add a standard interface for the drivetrain, so that it can be reused in both autonomous and operator control code.

**Describe the technicals that might be involved in implementing this**
We would likely need to have two classes: `Drivetrain` and `SkidSteerDrive`. The former would be a virtual class, and the second would be the derived class. 

The `Drivetrain` class would include all of the functions needed for a drivetrain, for example, going forward, backwards, turning, stopping, as well as using tank and arcade control schemes. However, they would all be virtual as implementation would vary depending on what type of drivetrain is being used.

The `SkidSteerDrive` class would simply implement all of these functions using motor calls.

By using the `Drivetrain` class, we can easily swap out the base drivetrain control in case of an unexpected switch.

An alternative would to just create a `SkidSteerDrive` class with all of the functions previously mentioned. Either works, though the former is more flexible.

**Additional context**
None.

### PR DESCRIPTION
**Pull request by hariskhawja**
Closes #1

### CODE REVIEW

File: include/driveSystems/SkidSteerDrive.h [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+
+#include "main.h"
+#include "drivetrain.h"
+
+class SkidSteerDrive : public Drivetrain {
+    public:
+        // tLeft --> top left motor
+        // tRight --> top right motor
+        // bLeft --> bottom left motor
+        // bRight --> bottom right motor
```
Comment (AritroSaha10) [inline include/driveSystems/SkidSteerDrive.h:18]: Can you change the function header comments to follow JavaDoc? It directly integrates with both IntelliSense and Doxygen Here's an example for the constructor:

```c++
/**
 * \brief Constructor for SkidSteerDrive class
 * 
 * \param tLeft top left motor
 * \param tRight top right motor
 * \param bLeft bottom left motor
 * \param bRight bottom right motor
*/
SkidSteerDrive(pros::Motor *tLeft, pros::Motor *tRight, pros::Motor *bLeft, pros::Motor *bRight);
```

File: include/driveSystems/SkidSteerDrive.h [hunk 2]
```diff
@@ -31,6 +31,9 @@ class SkidSteerDrive : public Drivetrain {
         // arcadebrakes to be explored and discussed
         void arcadeWithBrakes(double speed, double yaw, bool brakeLeft, bool brakeRight, double threshold = 0);
 
+        // to be reviewed; compiling issues with the vector?
```
Comment (AritroSaha10) [inline include/driveSystems/SkidSteerDrive.h:34]: Can you respond with the error that you're getting? I uncommented this part and it built normally.

**Review by AritroSaha10 — CHANGES_REQUESTED** [review]
Great! I just added a few things that should be changed. Also, I'm not sure if you were planning on doing this later, but since it's been a while since the last commit, could you also add some implementation for the `SkidSteerDrive` class?

File: include/driveSystems/SkidSteerDrive.h (hunk 1 above)
Comment (AritroSaha10) [reply include/driveSystems/SkidSteerDrive.h]: Fixed in f24b0e93410a736ffcad486443ad47c237794f98

**Comment by AritroSaha10** [comment]
I added some jsdoc-style comments in f24b0e93410a736ffcad486443ad47c237794f98. All that's left is implementing the functions defined in SkidSteerDrive.

**Comment by hariskhawja** [comment]
Oh thank you. I'll get to it as soon as possible

**Comment by AritroSaha10** [comment]
Looks good! Just one thing: function documentation is only necessary in header files, as they get auto-imported into the source files.

**Review by AritroSaha10 — APPROVED** [review]

**Pull request merged**